integration/helpers.go

```diff
@@ -124,6 +124,29 @@ func AssertEqualError(t testing.TB, expected mongo.CommandError, actual error) b
 	return assert.Equal(t, expected, a)
 }
 
+// AssertEqualAltError asserts that expected error is the same as actual
+// using altMessage for CommandError.Message if set and ignoring the Raw part.
+func AssertEqualAltError(t testing.TB, expected mongo.CommandError, altMessage string, actual error) bool {
+	t.Helper()
+
+	a, ok := actual.(mongo.CommandError)
+	if !ok {
+		return assert.Equal(t, expected, actual)
+	}
+
+	// raw part might be useful if assertion fails
+	require.Nil(t, expected.Raw)
+	expected.Raw = a.Raw
+
+	if !assert.ObjectsAreEqual(expected, a) { // compare with expected
+		// compare with expected with altMessage
+		expected.Message = altMessage
+		return assert.Equal(t, expected, a)
+	}
```

Comment:
Looks like my pseudocode, which just showed that `assert.ObjectsAreEqual` exists, was copied as is, and now this code is more complicated than it could be. For example, we essential have this:

```go
if !equal {
  // ...
}

assertEqual()
```

At the time we call `assertEqual`, we already know that objects are equal.

It all could be simplified to:

```go
if equal(expected, actual) {
  return
}

expected.Message = altMessage
if equal(expected, actual) {
  return
}

// fail with a good message mentioning both messages
```

Reply:
Updated

Reply:
Updated code now states: "if objects are equal, assert that they are equal"

Reply:
If we already checked that they are equal, we can just return true and don't make future readers scratch their heads :)

Reply:
Done.